x/outline-cli/main.go

```diff
@@ -0,0 +1,390 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-internal-sdk/network"
+	"github.com/Jigsaw-Code/outline-internal-sdk/network/dnstruncate"
+	"github.com/Jigsaw-Code/outline-internal-sdk/network/lwip2transport"
+	"github.com/Jigsaw-Code/outline-internal-sdk/transport"
+	"github.com/Jigsaw-Code/outline-internal-sdk/transport/shadowsocks"
+	"github.com/songgao/water"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+const OUTLINE_TUN_NAME = "outline233"
+const OUTLINE_TUN_IP = "10.233.233.1"
```

Comment:
I think we need a routing config type.

See
- https://developer.apple.com/documentation/networkextension/netunnelnetworksettings
- https://developer.apple.com/documentation/networkextension/nepackettunnelnetworksettings

for the settings that apply to iOS.